.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,46 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-dockerfile
+{
+	"name": "Base Image Development Environment",
+	"build": {
+		// Update the 'dockerFile' property if you aren't using the standard 'Dockerfile' filename.
+		"dockerfile": "Dockerfile"
+	},
+	"runArgs": [
+		"--init",
+		"--privileged"
+	],
+	"containerEnv": {
+		"GITHUB_API_TOKEN": "${localEnv:GITHUB_API_TOKEN}"
+	},
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Uncomment the next line to run commands after the container is created.
+	// "postCreateCommand": "cat /etc/os-release",
+
+	"features": {
+		"ghcr.io/devcontainers/features/common-utils:2": {},
+		"ghcr.io/devcontainers/features/docker-in-docker:2": {},
+		"ghcr.io/devcontainers/features/github-cli:1": {},
+		"ghcr.io/devcontainers-contrib/features/pre-commit:2": {}
```

Comment:
how is this configuring pre-commit? preventing commits if checks fail?

Reply:
Just use feature instead just installing it via apt get

Reply:
so no configuration at all?

Reply:
are you sure that this feature installation is doing no config?

Reply:
From sources: https://github.com/devcontainers-contrib/features/tree/main/src/pre-commit I didn't found something suspicious. From positive - it clean up after itself :)

Reply:
looking at this script I would highly recommend to not use the feature installation but add this to the dockerfile. It is installing some unneeded tools...
the only thing that is cleaned afterwards is that wget gets uninstalled...